### PR TITLE
feat: Improve Docker Compose state sync

### PR DESCRIPTION
**Pull request by luis02lopez**
Improve docker compose file to allow the volume to be in the same filesystem of the project and to use the Docker tag 'main' by default.

## Summary by CodeRabbit

- **Chores**
  - Updated Docker Compose configuration to use the latest version of `lavad` for enhanced compatibility and features.
  - Changed volume mount paths to local directories for easier debugging and data persistence.
  - Adjusted environment variables for improved configuration management.

### CODE REVIEW

**Review by Yaroms — DISMISSED** [review]

File: docker/docker-compose.state-sync.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ services:
 
   lava-node-init:
     # Update with the most recent supported version of lavad
-    image: ghcr.io/lavanet/lava/lavad:${LAVAD_VERSION:-v2.0.0}
+    image: ghcr.io/lavanet/lava/lavad:${LAVAD_VERSION:-main}
```
Comment (amitza) [inline docker/docker-compose.state-sync.yml:6]: `main` is by definition unstable, I would use `latest` or the latest stable version `v2.2.0`

**Review by amitza — CHANGES_REQUESTED** [review]

File: docker/docker-compose.state-sync.yml [hunk 2]
```diff
@@ -58,7 +58,7 @@ services:
 
   lava-node:
     # Update with the most recent supported version of lavad
-    image: ghcr.io/lavanet/lava/lavad:${LAVAD_VERSION:-v2.0.0}
+    image: ghcr.io/lavanet/lava/lavad:${LAVAD_VERSION:-main}
```
Comment (amitza) [inline docker/docker-compose.state-sync.yml:61]: https://github.com/lavanet/lava/pull/1557/files#r1672878897

**Review by amitza — CHANGES_REQUESTED** [review]

File: docker/docker-compose.state-sync.yml (hunk 1 above)
Comment (luis02lopez) [reply docker/docker-compose.state-sync.yml]: Done

**Review by amitza — APPROVED** [review]

**Comment by amitza** [comment]
Since I'm making major changes int the docker compose files, I'll add those changes in that PR

**Pull request closed without merging**